packages/adapter/addon/rest.js

```diff
@@ -1363,7 +1363,13 @@ export function fetchOptions(options, adapter) {
     } else {
       // NOTE: a request's body cannot be an object, so we stringify it if it is.
       // JSON.stringify removes keys with values of `undefined` (mimics jQuery.ajax).
-      options.body = JSON.stringify(options.data);
+      // If the data is not a POJO (it's a String, FormData, etc), we just set it.
+      // If the data is a string, we assume it's a stringified object.
+      if (Object.prototype.toString.call(options.data) === '[object Object]') {
```

Comment:
@nlfurniss - I'm curious, what cases does `Object.prototype.toString` cover that a `typeof` wouldn't? Would you mind updating this comment to explain that (and confirming that some of the new tests added specifically test that difference)?

Reply:
Originally it was to distinguish between an Object and an Array or other type that returns "object" from `typeof. But maybe we don't need to check that

Reply:
Because it doesn't catch the data structures (like FormData) that return `object` from `typeof`. Will add comment.

The test for FormData will fail if someone changes it to `typeof`

Reply:
it feels like `instanceof FormData` and `typeof x === 'string'` could be used to have a better guarantee here

Reply:
But its not just FormData that could be passed in, and it seemed like this was a lot nicer way of doing it than a large switch statement that may expand in the future